scipy/io/idl.py

```diff
@@ -31,7 +31,7 @@
 
 import struct
 import numpy as np
-from numpy.compat import asstr
+from scipy._lib._util import asstr
```

Comment:
You can remove this import. Then look at where it is used. In this case in one place:
```
    if length > 0:
        chars = _read_bytes(f, length)
        _align_32(f)
        chars = asstr(chars)
    else:
        chars = ''
```

Now, you look at the file mode with which `f` was opened. In this case that was `f = open(file_name, 'rb')`. So `f.read()` is guaranteed to give `bytes`. Hence you can change the code to:
```
    if length > 0:
        chars = _read_bytes(f, length).decode('latin1')
        _align_32(f)
    else:
        chars = ''
```

Then you run the tests to confirm - without `.decode('latin1')` they all fail, with the decode added they all pass.